scripts/kconfig/kconfig.py

```diff
@@ -57,6 +51,22 @@ def name_and_loc(sym):
             user_value = sym.user_value
         if user_value != sym.str_value:
             print('warning: {} was assigned the value "{}" but got the '
-                  'value "{}" -- check dependencies'
-                  .format(name_and_loc(sym), user_value, sym.str_value))
+                  'value "{}" -- check dependencies'.format(
+                      name_and_loc(sym), user_value, sym.str_value
+                  ),
+                  file=sys.stderr)
+
+
+# Turn the warning for malformed .config lines into an error
+for warning in kconf.warnings:
+    if "ignoring malformed line" in warning:
```

Comment:
A Kconfiglib generalization that would help future-proof this line is to associate a type with each of the "named" warnings (like "ignoring malformed line" has type WARN_TYPE_MALFORMED, etc.), and check that type here rather than matching the string exactly.

Those type tags could get set in each of `_warn_undef_assign`, `_warn_undef_assign_load`, and friends by specifying a kwarg to `_warn`. The `_kwarg` could default to WARN_TYPE_GENERAL or so.

That could also help going incrementally towards @SebastianBoe's ideas around GCC-style control of warnings by type.

Since Zephyr bundles its own Kconfiglib, that's not a blocker IMO.

Reply:
(This would require the list of warnings to become their own type of object, with the string text accessed by attribute, o/c).

Reply:
@mbolivar 
Was thinking you could just put the tag into the string itself, e.g. `"warning (w454): bad stuff"`, or something more readable. Think it might be overkilling it at this point though.